src/mongo/services/schemaService.ts

```diff
@@ -120,7 +120,7 @@ export class SchemaService {
 	private setSchemaForDocumentProperty(parent: string, property: string, document: any, schema: JSONSchema): void {
 		const scopedProperty = parent ? `${parent}.${property}` : property;
 		const value = document[property];
-		const type = Array.isArray(value) ? 'array' : typeof value;
+		const type = Array.isArray(value) ? 'array' : (value === null ? 'null' : typeof value);
```

Comment:
The logic is the same in both cases. Could you extract it out into a function (say `getMongoDocumentType`) that's called in both cases? I realize we didn't do it, but now that you've uncovered a bug, making this a function call will ensure the next use case won't hit the same bug you did. :)

Reply:
Do we need tests or testcases?

Reply:
I think that would be helpful.  We don't have any completion or intellisense tests, so I didn't want to pile that on this PR.  Let's pursue that here: 
https://github.com/microsoft/vscode-cosmosdb/issues/1167